src/main/java/picard/fingerprint/HaplotypeMap.java

```diff
@@ -41,21 +49,116 @@
+                }
+
+                if (!vc.isSNP()) {
+                    throw new IllegalStateException("Haplotype map VCF file must contain only SNPs: " + vc.toString());
+                }
+
+                if (!vc.hasAttribute(VCFConstants.ALLELE_FREQUENCY_KEY)) {
+                    throw new IllegalStateException("Haplotype map VCF Variants must have an '"+ VCFConstants.ALLELE_FREQUENCY_KEY + "' INFO field: " + vc.toString());
+                }
+
+                if (gc != null && gc.isPhased() && !gc.hasExtendedAttribute(VCFConstants.PHASE_SET_KEY)) {
+                    throw new IllegalStateException("Haplotype map VCF Variants' genotypes that are phased must have a PhaseSet ("+VCFConstants.PHASE_SET_KEY+")" + vc.toString());
+                }
+
+                if (gc != null && gc.isPhased() && !gc.isHet()) {
+                    throw new IllegalStateException("Haplotype map VCF Variants' genotypes that are phased must have a HET genotype" + vc.toString());
+                }
+
+                // Then parse them out
+                final String chrom = vc.getContig();
+                final int pos = vc.getStart();
+                final String name = vc.getID();
+
+                final byte ref = vc.getReference().getBases()[0];
+                final byte var = vc.getAlternateAllele(0).getBases()[0];
+
+                final double temp_maf = vc.getAttributeAsDouble(VCFConstants.ALLELE_FREQUENCY_KEY, 0D);
+                final boolean swapped = gc != null && !gc.getAllele(0).equals(vc.getReference());
+
+                final byte major, minor;
+                final double maf;
+
+                if (swapped) {
+                    major = var;
+                    minor = ref;
+                    maf = 1 - temp_maf;
+                } else {
+                    major = ref;
+                    minor = var;
+                    maf = temp_maf;
+                }
+
+                final String anchor = anchorFromVc(vc);
+
+                // If it's the anchor snp, start the haplotype
+                if (!anchorToHaplotype.containsKey(anchor)) {
+                    final HaplotypeBlock newBlock = new HaplotypeBlock(maf);
+                    anchorToHaplotype.put(anchor, newBlock);
+                }
+                final HaplotypeBlock block = anchorToHaplotype.get(anchor);
+                block.addSnp(new Snp(name, chrom, pos, major, minor, maf, null));
+            }
+
+            // And add them all now that they are all ready.
+            anchorToHaplotype.values().forEach(this::addHaplotype);
+        }
+    }
+
+    static private String anchorFromVc(final VariantContext vc) {
+        final Genotype genotype = vc.getGenotype(0);
+
+        if (genotype == null || !genotype.hasExtendedAttribute(VCFConstants.PHASE_SET_KEY) ) {
+           return "Synthetic_"+vc.getContig()+"_"+vc.getStart();
```

Comment:
spaces between + 
also all magic string should be constants "Synthetic" and "PhaseSet"